mirrulations-client/src/mirrclient/disk_saver.py

```diff
@@ -70,3 +70,42 @@ def is_duplicate(self, existing, new):
     def check_for_duplicates(self, path, data, i):
         if self.is_duplicate(self.open_json_file(path), data) is False:
             self.save_duplicate_json(path, data, i)
+
+    def get_extracted_text_attachments_dir(self, path):
+        # remove file name from path
+        path = path.rsplit("/", 1)[0]
+        return path.replace('comments_extracted_text/pdfminer',
+                            'comments_attachments/').replace('text', 'binary')
+
+    def generate_metadata_for_extraction(self, path):
```

Comment:
This function should probably be part of the extractor. The saver class's job is to save, not generate a file.